tests/utils/helpers.rs

```diff
@@ -592,6 +603,20 @@ impl TestWallet {
         self.wallet.wallet().seal_close_method()
     }
 
+    pub fn mine_tx(&self, txid: &Txid, resume: bool) {
+        let mut attempts = 10;
+        loop {
+            mine(resume);
+            if self.get_tx_height(txid).is_some() {
```

Comment:
Just double-checking: for a mempool tx `get_tx_height` returns `None`, right? Not `Some(0)`

Reply:
That method calls the `resolve_pub_witness_ord` method from the resolver which returns `WitnessOrd::Tentative` for mempool TXs. Being the `get_tx_height` method defined as:
```rust
    pub fn get_tx_height(&self, txid: &Txid) -> Option<u32> {
        match self
            .get_resolver()
            .resolve_pub_witness_ord(XWitnessId::Bitcoin(*txid))
            .unwrap()
        {
            WitnessOrd::Mined(witness_pos) => Some(witness_pos.height().get()),
            _ => None,
        }
    }
```
I'd say that for mempool TXs it returns `None`.